### PR TITLE
[BugFix] prune complex type need deal with scan without project

### PR DESCRIPTION
**Pull request by zombee0**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java|11|11|100.00%|[]|

**Review by Seaven — DISMISSED** [review]

**Review by Smith-Cruise — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java [hunk 1]
```diff
@@ -117,6 +121,24 @@ public Void visitPhysicalJoin(OptExpression optExpression, PruneComplexTypeUtil.
...
         }
+
+        @Override
+        public Void visitPhysicalScan(OptExpression optExpression, PruneComplexTypeUtil.Context context) {
+            PhysicalScanOperator physicalScanOperator = (PhysicalScanOperator) optExpression.getOp();
+            Projection projection = optExpression.getOp().getProjection();
+            if (projection == null) {
+                for (ColumnRefOperator columnRefOperator : physicalScanOperator.getOutputColumns()) {
+                    if (columnRefOperator.getType().isMapType() || columnRefOperator.getType().isStructType()) {
+                        ComplexTypeAccessPath complexTypeAccessPath =
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java:132]: context.add(columnRefOperator, columnRefOperator) is better?

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java]: Done

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java|8|8|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**